### PR TITLE
fix: tokenAssociate throws IndexOutOfBoundsException (#12453)

### LINKED ISSUE #12440: `tokenAssociate` throws `IndexOutOfBoundsException`
### Description

`tokenAssociate` throws `IndexOutOfBoundsException` when passing no tokens param to the request body.

### Steps to reproduce

-

### Additional context

_No response_

### Hedera network

other

### Version

0.48

### Operating system

macOS

### PR DESCRIPTION
**Pull request by JivkoKelchev**
**Description**:

This PR adds an additional check to ensure that the newly created token relation list is not empty. This way, we prevent IndexOutOfBoundsException when passing no tokens param to the request body.

**Related issue(s)**:

Fixes #12440

**Notes for reviewer**:

This is a quick fix and it match mono behaviour, and return is SUCCES, but we may need to return a proper error status when tokens are empty.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java [hunk 1]
```diff
@@ -173,19 +173,22 @@ protected void createAndLinkTokenRels(
...
-                .headTokenId(firstOfNewTokenRels.tokenId())
-                // and also update the account's total number of token associations
-                .numberAssociations(account.numberAssociations() + newTokenRels.size())
-                .build();
-
-        // Save the results
-        accountStore.put(updatedAcct);
-        newTokenRels.forEach(tokenRelStore::put);
+        // FUTURE - We may need to return a proper error status when tokens are empty
+        if (!newTokenRels.isEmpty()) {
```
Comment (netopyr) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java:177]: I think we have to include the call of `linkTokenRels()` (one line above). If `newTokenRels` is empty, the remove-call in line 220 fails, doesn't it?

**Comment by thomas-swirlds-labs** [comment]
Do we not want to include the `associateTokensNegativeScenarios` test in release/0.48 as well ?

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java (hunk 1 above)
Comment (JivkoKelchev) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java:177]: Good catch. I was using an account without previous relations, so the remove was never executed.
I will create a new PR to develop and cherry-pick here. 
Thanks

**Comment by JivkoKelchev** [comment]
> Do we not want to include the `associateTokensNegativeScenarios` test in release/0.48 as well ?

I got merge conflicts when I cherry-picked it, because of this test.
And since it was part of another [PR](https://github.com/hashgraph/hedera-services/pull/12163) that was not included in release/0.48, I decided to accept changes from 0.48 (remove the test)

**Review by netopyr — APPROVED** [review]
LGTM - thanks @JivkoKelchev

**Review by thomas-swirlds-labs — APPROVED** [review]
LGTM! tyvm @JivkoKelchev

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @JivkoKelchev !

**Pull request merged**